app/views/events/index.html.haml

```diff
@@ -5,7 +5,10 @@
   - events.each do |event|
     %div
       = link_to event.title, event_path(event)
-=link_to 'Create new event', new_event_path, :class => 'btn btn-flat create-event'
+- if user_signed_in?
+  =link_to 'Create new event', new_event_path, :class => 'btn btn-flat create-event'
+- else
```

Comment:
The else is unnecessary and leads to problems... So line 10 and 11 should be deleted.